axum/src/routing/route.rs

```diff
@@ -42,13 +42,27 @@ impl<E> Route<E> {
         )))
     }
 
+    /// Variant of [`Route::call`] that takes ownership of the route to avoid cloning.
+    pub(crate) fn call_owned(self, req: Request<Body>) -> RouteFuture<E> {
+        let req = req.map(Body::new);
+        RouteFuture::from_future(self.oneshot_inner_owned(req))
```

Comment:
When comparing with `main`, it would now be `self.oneshot_inner_owned(req).not_top_level()`.

I'm not sure about the `.not_top_level()`. And there are no tests checking if this is necessary or not.
Do you think we could test this?

Reply:
This is a note for me.
The current `.not_top_level()` is covered by `axum/src/routing/tests/mod.rs:1086`